### PR TITLE
Alter type saving logic in CreateLoad

### PR DESCRIPTION
**Pull request by advikkabra**
Previously, `type_copy` was saved only when it did not equal `type`, but it could be possible that `type_copy` is a pointer and so `type` would be a pointer as well.

Example:
```
type_copy = ptr
type = ptr->getPointerTo() // also a ptr
```

### CODE REVIEW

File: src/libasr/codegen/llvm_utils.cpp [hunk 1]
```diff
@@ -1633,6 +1633,7 @@ namespace LCompilers {
         return builder->CreateLoad(t2, x);
 #else
         llvm::Type *type = nullptr, *type_copy = nullptr;
+        bool flag = false;
```
Comment (gxyd) [inline src/libasr/codegen/llvm_utils.cpp:1636]: can we use a more descriptive name than `flag`? The keyword `flag` doesn't give much idea into what it's purpose is, and is it supposed to be set as true/false.

Comment (advikkabra) [reply src/libasr/codegen/llvm_utils.cpp]: Yes I'll do that.

**Review by gxyd — APPROVED** [review]
LGTM!

**Comment by Thirumalai-Shaktivel** [comment]
Thank you @advikkabra!

**Pull request merged**